### PR TITLE
Support completion inserter

### PR DESCRIPTION
**Pull request by karenzheung**
_Pull requests into Flux require the following. Submitter should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable (**N/A**) and :white_check_mark:. Reviewers should complete every item on the bulleted list when reviewing._

## Submitter:
ASCODCP-1908

After you insert an incomplete @condition you can go back and complete it several ways
1.) Click on the incomplete @condition to select value from context portal
2.) arrow key into the @condition or set cursor to right of it to bring up context portal 

**Running the Application**

- [x] Manually tested in Chrome (primary browser for testing)
- [ ] Manually tested in IE/Safari (verify app loads and basic feature testing)
- [ ] Manually tested in Chrome in iPad mode (at minimum, viewing patient record)

**Task**

- [x] Branch implements task as expected
- [x] Key application features are updated as needed for task and tested (see: [Key Application Features](https://github.com/FluxNotes/flux/wiki/Key-Application-Features))
- [x] Task specific definition of done (as described in the Jira task) is met, if applicable.

**Documentation**

- [ x] This pull request describes why these changes were made
- [x ] Recognizes any potential shortcomings/bugs in the description 
- [x ] Pre release script is updated on the [Pre Release Testing Script wiki page](https://github.com/FluxNotes/flux/wiki/Pre-Release-Testing-Script) (add/remove single features to test, do not run through full script)
- [x ] Update the list of key features with new features on the [Key Application Features wiki page](https://github.com/FluxNotes/flux/wiki/Key-Application-Features)
- [x ] Document any new APIs/extension points
- [x ] Additional technical components and libraries are documented and added to [wiki](https://github.com/FluxNotes/flux/wiki/About-Flux-Notes) as needed

**Code Quality**

- [ x] Code style guide is followed (see: [Code Style Guide](https://github.com/FluxNotes/flux/wiki/JavaScript-and-HTML-Code-Style-Guide))
- [ x] Code is commented

**Tests**

- [ x] Existing tests passed
- [ x] Added backend tests for new functionality added

## Reviewers:

- Check that code is maintainable and reusable. Code reuses existing code and infrastructure where appropriate.
- Check the PR and code accomplishes the task's purpose.
- Check that new tests appropriately test the new code, including edge cases. Other existing tests pass.
- Verify Pull Request checklist is correctly completed by submitter.
- Try to break the code

### CODE REVIEW

**Comment by mgramigna** [comment]
[MG1] @condition is no longer showing up as incomplete for me on this branch, it is still fine on the other branch

**Comment by mgramigna** [comment]
I think this branch needs the changes from your other PR

**Comment by karenzheung** [comment]
I made this branch off the branch for my other PR. It shows on incomplete on my end.

**Comment by mgramigna** [comment]
Yeah my git log actually seems fine.. I'll report back

**Comment by mgramigna** [comment]
FYI: I updated this branch to be rebased off of `shortcut-entry-overhaul` instead of its previous base branch as that branch was deleted in #627. Updated the base branch of the PR accordingly

File: src/notes/FluxNotesEditor.jsx [hunk 1]
```diff
@@ -386,7 +386,7 @@ class FluxNotesEditor extends React.Component {
         return this.lastPosition;
     }
 
-    openPortalToSelectValueForShortcut(shortcut, needToDelete, transform) {
+     openPortalToSelectValueForShortcut(shortcut, needToDelete, transform) {
```
Comment (Dtphelan1) [inline src/notes/FluxNotesEditor.jsx]: Can we fix this?

**Comment by Dtphelan1** [comment]
Good changes 👍 

[DP1] (open to discussion) When I click 'esc' I expect my cursor to be back in focus after the condition. That doesn't seem to be happening on my end.

**Comment by mgramigna** [comment]
[MG2] Typing @condition and then leaving it as incomplete, closing the note, then trying to re-open causes the app to crash 

Assuming the text of this incomplete shortcut is saved in a way that we don't know how to re-load it into the note?

**Comment by mgramigna** [comment]
[KZ1/MG3] Cursor does not blur when opening the portal in a special edge case:

1) Insert an incomplete @condition
2) Hit Esc to close the portal
3) Continue typing some text
4) Either left arrow key or click next to the incomplete @condition to re-open the portal
5) Blur doesn't happen :(

Karen and I are looking into this now

**Comment by mgramigna** [comment]
Karen and I resolved MG2, with help from Dan :)

**Comment by jafeltra** [comment]
I'm going to close this PR temporarily, as the subtask it is addressing is not in our current sprint. When the subtask is in the sprint, this PR should be reopened and all comments should be addressed then.

Subtask: https://jira.mitre.org/browse/ASCODCP-1908

NOTE: The branch for this (support-completion-inserter) should not be deleted, as this will be reopened and addressed eventually.

**Pull request closed without merging**